src/lib/Hatchery.js

```diff
@@ -998,8 +998,8 @@ class AutomationHatchery
                         && (evolution?.stone == GameConstants.StoneType.Key_stone)
                         && PokemonHelper.calcNativeRegion(evolution.evolvedPokemon) <= player.highestRegion())
                     {
-                        // Only consider mega evolution with an incomplete mega evolution requirement
-                        return !evolution.restrictions?.filter(e => Automation.Utils.isInstanceOf(e, "MegaEvolveRequirement"))[0]?.isCompleted();
+                        // Cant check for mega stone here, as the pokemon might not have it yet
+                        return partyPokemon.attack < partyPokemon.baseAttack * 100;
```

Comment:
Thanks for the clarification, here is the implementation that I would prefer then : 
```suggestion
                        const megaEvolveReqs = evolution.restrictions?.filter(e => Automation.Utils.isInstanceOf(e, "MegaEvolveRequirement"));

                        // Don't consider pokemon that does not have a mega evolution
                        if (megaEvolveReqs.length < 1)
                        {
                            return false;
                        }

                        // Only consider pokemon with an incomplete mega-stone requirement (buid it since the pokemon might not have unlocked it yet)
                        return !(new MegaStone(partyPokemon.id, partyPokemon.baseAttack, partyPokemon._attack).canEvolve());
```